keyboards/preonic/keymaps/laurentlaurent/keymap.c

```diff
@@ -0,0 +1,603 @@
+        __LYB__,  KC_ESC,  KC_DEL,    KC_ENT,   KC_BSPACE, KC_SPC,   KC_SPC,   LMW_LT,   LMW_BT,   LMW_TP,   LMW_RT,  LLSWIT
+    ),
+
+    /* Layer Switcher
+     * ,-----------------------------------------------------------------------------------.
+     * | Esc  | XXXX | XXXX | XXXX | XXXX | XXXX | XXXX | XXXX | XXXX | XXXX | XXXX | XXXX |
+     * |------+------+------+------+------+------+------+------+------+------+------+------|
+     * |Brite | XXXX | XXXX | XXXX | XXXX | XXXX | XXXX | XXXX | XXXX | XXXX | XXXX | XXXX |
+     * |------+------+------+------+------+-------------+------+------+------+------+------|
+     * | XXXX | XXXX | XXXX | XXXX | XXXX | GAME | XXXX | XXXX | XXXX | XXXX | NAV  | XXXX |
+     * |------+------+------+------+------+------|------+------+------+------+------+------|
+     * |      | XXXX | XXXX | XXXX | XXXX | QMAC | XXXX | MUSC | XXXX | XXXX | XXXX | XXXX |
+     * |------+------+------+------+------+------+------+------+------+------+------+------|
+     * | XXXX | XXXX | NUM  | XXXX | XXXX |    XXXXX    | XXXX | XXXX | XXXX | XXXX | !!!! |
+     * `-----------------------------------------------------------------------------------'
+     */
+    [_LYSWT] = LAYOUT_preonic_grid( \
+        KC_ESC ,   XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,  	XXXXXXX,  XXXXXXX,   XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,
+        BACKLIT,   XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,  	XXXXXXX,  XXXXXXX,   XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,
+        XXXXXXX,   XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,  TO(_GAMING), XXXXXXX,  XXXXXXX,   XXXXXXX,  XXXXXXX,  TO(_NAV), XXXXXXX,
+        XXXXXXX,   XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,  LTO_BS,   	XXXXXXX, TO(_MUSIC), XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,
+        XXXXXXX,   XXXXXXX,  TO(_NUM), XXXXXXX,  XXXXXXX,  XXXXXXX,  	XXXXXXX,  XXXXXXX,   XXXXXXX,  XXXXXXX,  XXXXXXX,  _______
+    ) //, //Don't forget to add the comma if going to add more layers here
+};
+
+static uint16_t key_timer;
+static uint16_t timer_thresh = 200;
+bool            process_record_user(uint16_t keycode, keyrecord_t *record) {
+    // For dynamic macros
+    if (!process_record_dynamic_macro(keycode, record)) {
+// Play sound on Macro stop
+#ifdef AUDIO_ENABLE
+        switch (keycode) {
+            case DYN_REC_STOP:
+                if (record->event.pressed) {
+                    PLAY_SONG(tone_macro_record_stop);
+                }
+                return false;
+                break;
+        }
+#endif
+        return false;
+    }
+
+    switch (keycode) {
+        case QWERTY:
+            if (record->event.pressed) {
+                set_single_persistent_default_layer(_QWERTY_MAC);
+            }
+            return false;
+            break;
+        case QWWIN:
+            if (record->event.pressed) {
+                set_single_persistent_default_layer(_QWERTY_WIN);
+            }
+            return false;
+            break;
```

Comment:
It looks like the only changes between these two layers are the control and gui/cmd keys?  

If that's the case, you may want to use the control-command swap feature instead:
https://docs.qmk.fm/#/feature_bootmagic?id=keycodes

Even if bootmagic is disabled, these keycodes will still work.  And the `CG_*` keycode will swap these keys.

Reply:
I considered using the swap keycode, but decided against it:
My other layers relies on the Ctrl/GUI staying at the same spots (like in Lower or Raise)

I wasn't sure abt something: if I made my _QWERTY_WIN layer into mostly KC_TRANS's (except for the keys that matter), if I set the _QWERTY_WIN layer into the default persistent layer, would it still fall through into _QWERTY_MAC?

Is there any way I can "get" the default layer, for some more advanced custom keys? What would they be named?

i.e. Volume Up is the Windows volume up if default = _QWERTY_WIN, Mac volume up if default = _QWERTY_MAC.

Thanks

Reply:
The volume keycodes under the [Media Keys](https://docs.qmk.fm/#/keycodes_basic?id=media-keys) section will work across all OSes :) you don't need to set up separate layers just for that.